selfdrive/common/clutil.cc

```diff
@@ -0,0 +1,212 @@
+#include "clutil.h"
+
+#include <assert.h>
+#include <inttypes.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <memory>
+#include <vector>
+#include "util.h"
+#include "utilpp.h"
+namespace {  // helper functions
+
+std::string get_platform_info(cl_platform_id platform, cl_platform_info param_name) {
+  size_t size = 0;
+  CL_CHECK(clGetPlatformInfo(platform, param_name, 0, NULL, &size));
+  std::string ret;
+  ret.resize(size, '\0');
+  CL_CHECK(clGetPlatformInfo(platform, param_name, size, &ret[0], NULL));
+  return ret;
+}
+
+void cl_print_info(cl_platform_id platform, cl_device_id device) {
+  std::string info = get_platform_info(platform, CL_PLATFORM_VENDOR);
+  printf("vendor: '%s'\n", info.c_str());
```

Comment:
`std::cout << "vendor: " << info << std::endl;`?

Reply:
Or just skip the temp variable altogether:
`std::cout << "vendor: " << get_platform_info(platform, CL_PLATFORM_VENDOR) << std::endl;

Reply:
cool